generator.js

```diff
@@ -1,15 +1,15 @@
 module.exports = api => {
   const isTs = api.entryFile.endsWith('.ts')
-  const usesRouter = Boolean(require(api.resolve('package.json')).dependencies['vue-router'])
-  const projectName = require(api.resolve('package.json')).name;
+  const packageJson = require(api.resolve('package.json'));
+  const usesRouter = Boolean(packageJson.dependencies && packageJson.dependencies['vue-router']);
 
   api.render({
     [api.entryFile]: './template/src/main.js',
     './src/set-public-path.js': './template/src/set-public-path.js',
   }, {
     isTs,
     usesRouter,
-    appName: projectName,
+    appName: packageJson.name,
```

Comment:
It's possible for package.jsons to not have a `name` property in them - should we update this to have a default name when that is the case?

```suggestion
    appName: packageJson.name || 'appName',
```

Reply:
Yea, I think having a default is sensible.